ybd/repos.py

```diff
@@ -160,11 +155,26 @@ def mirror_has_ref(gitdir, ref):
         return out == 0
 
 
+def reinit_as_bare(gitdir):
+    with app.chdir(gitdir):
+        for f in os.listdir("."):
+            if os.path.isdir(f):
+                shutil.rmtree(f)
+            else:
+                os.remove(f)
+
+        call(['git', 'init', '--quiet', '--bare'])
+
+
 def update_mirror(name, repo, gitdir):
     with app.chdir(gitdir), open(os.devnull, "w") as fnull:
+        bare_config = check_output(['git', 'config', 'core.bare']).strip()
+        if bare_config == "false":
+            app.log(name, 'Reinitialising %s as bare repo' % repo)
+            reinit_as_bare(gitdir)
         app.log(name, 'Refreshing mirror for %s' % repo)
-        if call(['git', 'remote', 'update', 'origin'], stdout=fnull,
-                stderr=fnull):
+        if call(['git', 'fetch', get_repo_url(repo), '+refs/*:refs/*'],
```

Comment:
This is technically a change in behaviour,
since git remote update origin with remote.origin.mirror set may also delete local refs,
while this becomes an ever expanding set.